OpenRA.Game/Actor.cs

```diff
@@ -135,7 +135,7 @@ Rectangle DetermineBounds()
 			var size = (si != null && si.Bounds != null) ? new int2(si.Bounds[0], si.Bounds[1]) :
 				TraitsImplementing<IAutoSelectionSize>().Select(x => x.SelectionSize(this)).FirstOrDefault();
 
-			var offset = -size / 2;
+			var offset = TraitsImplementing<IAutoSelectionSize>().Select(x => x.SelectionOffset(this)).FirstOrDefault() - size / 2;
```

Comment:
Rather than potentially looking up `IAutoSelectionSize` twice, I think you should draw out a `var autoSelectionSize = TraitsImplementing<IAutoSelectionSize>();` to befote line 135 and only perform the `.Select` here and in `size`.